compiler/llvm/clangUtil.cpp

```diff
@@ -2628,12 +2629,23 @@ llvm::Type *LayeredValueTable::getType(StringRef name) {
 
       // Convert it to an LLVM type.
       store->u.type = codegenCType(store->u.cTypeDecl);
+      const clang::Type *type = store->u.cTypeDecl->getTypeForDecl();
+      if (type != NULL) {
+        store->isUnsigned = type->isUnsignedIntegerOrEnumerationType();
+      }
       return store->u.type;
     }
   }
   return NULL;
 }
 
+bool LayeredValueTable::getIsUnsigned(StringRef name) {
```

Comment:
I probable would have just added a `bool& isUnsigned`  argument to `LayeredValueTable::getType` but this is also OK.

Reply:
That would require adding an argument to every `LayeredValueTable::getType` call, even when they don't care about `isUnsigned`. How about a `bool*` instead with a default `NULL` value?